buildspec_sync.yml

```diff
@@ -5,16 +5,41 @@ phases:
       python: 3.7
   pre_build:
     commands:
-      - echo Sync latest image from Dockerhub
+      - echo Sync latest image from Amazon ECR Public Gallery
       - curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
       - unzip awscliv2.zip
-      - ./aws/install
+      - which aws
+      - ls -l /root/.pyenv/shims/aws
+      - ./aws/install --bin-dir /root/.pyenv/shims --install-dir /root/.pyenv/shims --update
   build:
     commands:
       # Enforce STS regional endpoints
       - export AWS_STS_REGIONAL_ENDPOINTS=regional
-      - './scripts/publish.sh cicd-publish ${AWS_REGION}'
+      
+      - './scripts/publish.sh cicd-publish ${AWS_REGION} stable' 
 
+      # Publish stable tag to Dockerhub
+      - './scripts/publish.sh cicd-publish ${AWS_REGION} public-dockerhub'
+
+       # Assume role to publish, get the credentials, and set them as environment variables
+      - |
+        if [ "${PUBLISH_ROLE_ARN_PUBLIC_ECR}" != "" ]; then
+          CREDS=`aws sts assume-role --role-arn ${PUBLISH_ROLE_ARN_PUBLIC_ECR} --role-session-name publicECR`
+          export AWS_ACCESS_KEY_ID=`echo $CREDS | jq -r .Credentials.AccessKeyId`
+          export AWS_SECRET_ACCESS_KEY=`echo $CREDS | jq -r .Credentials.SecretAccessKey`
+          export AWS_SESSION_TOKEN=`echo $CREDS | jq -r .Credentials.SessionToken`
+        fi
+       # Publish stable tag to Public ECR
+      - './scripts/publish.sh cicd-publish ${AWS_REGION} public-ecr'
```

Comment:
may be add a comment that these publish commands only do something if the region is us-west-2? When the sync task runs in bahrain they will have no affect right? We use the same buildspec file for all sync tasks I think.